apps/rush-lib/package.json

```diff
@@ -30,7 +30,6 @@
     "@rushstack/ts-command-line": "workspace:*",
```

Comment:
> ## Summary
> 
> The `chokidar` package provides a bunch of signal normalization that is redundant for the watch capabilities of rush, because the actual set of changed files is used purely as a boolean signal that is then handed off to the `PackageChangeAnalyzer` to determine what actually changed.
> 
> This PR removes `chokidar` and replaces it with native `fs.watch` calls.

At HBO we originally used `fs.watch` for watching and found it to be extremely slow.  As in, flaming MacBooks twirling and flying off of people's desks like a Class C aerial spinner firework.  😋

[chokidar](https://www.npmjs.com/package/chokidar) implements some important optimizations and is the gold standard for file watching.  Are we sure we shouldn't be using it?

Reply:
The only thing that might matter to us in `chokidar` is using `fsevents` directly instead of `fs.watch` on MacOS. Every other "optimization" is completely counterproductive for our use case, since the file system watcher is *not* the authority on what files changed, which is normally what `chokidar` adds these various layers to do. Per official node documentation, NodeJS `fs.watch` uses FSEvents when watching directories, which are the only thing we are watching.

I'd be open to a performance comparison between this build and the one with chokidar on MacOS, but my expectation is that this version will actually be faster, since it is doing less work. Also gets rid of the annoying optional dependency error during installation on non-MacOS.

Reply:
It would be smart to do some kind of measurements on Linux/Windows/Mac to double-check that `fs.watch` isn't regressing performance somehow.  Specifically with regards to [this optimization](https://www.npmjs.com/package/chokidar):

> On MacOS, chokidar by default uses a native extension exposing the Darwin FSEvents API. This provides very efficient recursive watching compared with implementations like kqueue available on most *nix platforms.

In the longer term we're anticipating a more scalable design where a centralized Rush process watches the entire monorepo folder tree, and then Jest/Webpack/etc get notified by Rush (rather than watching for file events themselves).  So although Rush's watching requirements are somewhat limited right now, it made me think that we may need more of Chokidar's advanced features later down the road.  

However, @dmichon-msft has pointed out that Rush's primary optimization is to use PackageChangeAnalyzer to leverage Git's change detection.  The `fs.watch` part only needs to provide rough granularity notifications which trigger the PackageChangeAnalyzer.  That sounds like a persuasive argument that we can safely eliminate Chokidar. 👍